Submit clarification with problem label

deboer-tim opened the pull request:
When submitting a clarification this verifies --problem if it exists, and allows it to be the problem id ("hobson"), label ("H"), or name ("Hobson's Trains"). Adding the name also allows problem name to be used when submitting.

nickygerritsen commented on contest.go:
```diff
@@ -44,6 +51,48 @@ func (b basicAuthTransport) RoundTrip(request *http.Request) (*http.Response, er
 }
 
+// parse a relative time (regex: "(-)?(h)*h:mm:ss(.uuu)?") and convert to integer milliseconds
+func parseRelTime(relTime string) int64 {
```
Instead of int64, we could use the build in Duration type. It will make the below method easier. But we can do it later (I can try it)

deboer-tim replied:
I didn't know that type existed :) and I don't see how we'd initialize it except by creating a ms string... but agree it would make the following method better.

nickygerritsen replied:
If you have the number of seconds you can do
```
var d = timeinsecs * time.Second
```
IIRC

nickygerritsen replied:
Which you can probably also use to add the separate parts without having “magic” 3600’s. I will try it soon

nickygerritsen commented on contest.go:
```diff
@@ -126,15 +175,23 @@ func main() {
-			fmt.Println("Contests found successfully!", contests)
-			fmt.Printf("Contests: %+v", contests)
+			fmt.Printf("Contests (%d):\n", len(contests))
+			for i := range contests {
```
We could make contest adhere to fmt.gostringer: https://golang.org/pkg/fmt/#GoStringer

deboer-tim replied:
FWIW, I tried that first but it was actually longer and I wasn't sure we'd reuse it elsewhere in exactly the same output format.

deboer-tim committed:
Submit clarification with problem label

When submitting a clarification this verifies --problem if it exists, and allows it to be the problem id ("hobson"), label ("H"), or name ("Hobson's Trains"). Adding the name also allows problem name to be used when submitting.
